cocotb/regression.py

```diff
@@ -305,23 +305,32 @@ class TestFactory(object):
     the test description) includes the name and description of each generator.
     """
 
-    def __init__(self, test_function, *args):
+    def __init__(self, test_function, name_postfix="", *args, **kwargs):
```

Comment:
Thanks for finding this, I agree that this may be a potential problem.

It is ok to have the argument on generate_tests() instead, I do not really
prefer one over the other in this case.

I am using it for permuting the test_function with different
parameter-sets, as combining all permutations would lead to too long
test-time. prefix and postfix are better names as well, when we use them as
arguments to generate_tests I think we can avoid the longer name_postfix
name. Good idea!

Sigve

On Fri, Jan 16, 2015 at 9:49 PM, Chris Higgs <notifications@github.com>
wrote:

> In cocotb/regression.py
> <https://github.com/potentialventures/cocotb/pull/192#discussion-diff-23108373>
> :
>
> > @@ -305,23 +305,32 @@ class TestFactory(object):
> >      the test description) includes the name and description of each generator.
> >      """
> >
> > -    def __init__(self, test_function, *args):
> > +    def __init__(self, test_function, name_postfix="", *args, **kwargs):
>
> I think this might cause problems. I'm not convinced Python will like
> having a named kwarg before *args and then a **kwargs. I forget which
> version but previously I've had to do something like this:
>
> def my_func(*args, **kwargs):
>     name_postfix = kwargs.get("name_postfix", None)
>
> See also this StackOverflow question
> <http://stackoverflow.com/questions/14003939/python-function-args-and-kwargs-with-other-specified-keyword-arguments>
>
> I like the idea of passing **kwargs to the test function. I suggest we
> make postfix (and maybe prefix) keyword arguments to generate_tests()
> instead, thus avoiding any confusion.
>
> Would that work for your use-case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/potentialventures/cocotb/pull/192/files#r23108373>.
>